lib/Controller/ChatController.php

```diff
@@ -841,12 +850,17 @@ public function getObjectsSharedInRoom(string $objectType, int $lastKnownMessage
 		return $response;
 	}
 
-	protected function getMessagesForRoom(Room $room, array $messageIds): array {
+	private function getMessagesForRoom(Room $room, array $messageIds): array {
 		$comments = $this->chatManager->getMessagesById($room, $messageIds);
 
 		$messages = [];
 		foreach ($comments as $comment) {
+			if (!$this->messageParser->fileOfMessageExists($comment->getMessage())) {
```

Comment:
```suggestion
			if ($this->messageParser->isSharedFile($comment->getMessage()) && !$this->messageParser->fileOfMessageExists($comment->getMessage())) {
```

Reply:
Won't we have only messages as shared file here?

Reply:
see above